test/integration/master/audit_test.go

```diff
@@ -164,62 +173,128 @@ rules:
-	if err != nil {
-		t.Fatalf("Failed to create audit policy file: %v", err)
-	}
-	defer os.Remove(policyFile.Name())
-	if _, err := policyFile.Write(auditPolicy); err != nil {
-		t.Fatalf("Failed to write audit policy file: %v", err)
-	}
-	if err := policyFile.Close(); err != nil {
-		t.Fatalf("Failed to close audit policy file: %v", err)
-	}
+func testAudit(t *testing.T, version string, level auditinternal.Level, enableMutatingWebhook bool) {
+	t.Run(fmt.Sprintf("%s.%s.%t", version, level, enableMutatingWebhook), func(t *testing.T) {
+		var url string
+		var closeFunc func()
+		if enableMutatingWebhook {
+			url, closeFunc = newWebhookServer(t)
+		}
 
-	// prepare audit log file
-	logFile, err := ioutil.TempFile("", "audit.log")
-	if err != nil {
-		t.Fatalf("Failed to create audit log file: %v", err)
-	}
-	defer os.Remove(logFile.Name())
-
-	// start api server
-	result := kubeapiservertesting.StartTestServerOrDie(t, nil,
-		[]string{
-			"--audit-policy-file", policyFile.Name(),
-			"--audit-log-version", version,
-			"--audit-log-mode", "blocking",
-			"--audit-log-path", logFile.Name()},
-		framework.SharedEtcd())
-	defer result.TearDownFn()
-
-	kubeclient, err := kubernetes.NewForConfig(result.ClientConfig)
-	if err != nil {
-		t.Fatalf("Unexpected error: %v", err)
-	}
+		// prepare audit policy file
+		auditPolicy := strings.Replace(auditPolicyPattern, "{version}", version, 1)
+		auditPolicy = strings.Replace(auditPolicy, "{level}", string(level), 1)
+		policyFile, err := ioutil.TempFile("", "audit-policy.yaml")
+		if err != nil {
+			t.Fatalf("Failed to create audit policy file: %v", err)
+		}
+		defer os.Remove(policyFile.Name())
+		if _, err := policyFile.Write([]byte(auditPolicy)); err != nil {
+			t.Fatalf("Failed to write audit policy file: %v", err)
+		}
+		if err := policyFile.Close(); err != nil {
+			t.Fatalf("Failed to close audit policy file: %v", err)
+		}
 
-	// perform configmap operations
-	configMapOperations(t, kubeclient)
+		// prepare audit log file
+		logFile, err := ioutil.TempFile("", "audit.log")
+		if err != nil {
+			t.Fatalf("Failed to create audit log file: %v", err)
+		}
+		defer os.Remove(logFile.Name())
 
-	// check for corresponding audit logs
-	stream, err := os.Open(logFile.Name())
-	if err != nil {
-		t.Fatalf("Unexpected error: %v", err)
+		// start api server
+		result := kubeapiservertesting.StartTestServerOrDie(t, nil,
+			[]string{
+				"--audit-policy-file", policyFile.Name(),
+				"--audit-log-version", version,
+				"--audit-log-mode", "blocking",
+				"--audit-log-path", logFile.Name()},
+			framework.SharedEtcd())
+		defer result.TearDownFn()
+
+		kubeclient, err := kubernetes.NewForConfig(result.ClientConfig)
+		if err != nil {
+			t.Fatalf("Unexpected error: %v", err)
+		}
+
+		if enableMutatingWebhook {
+			if err := createV1beta1MutationWebhook(kubeclient, url+"/mutation"); err != nil {
+				t.Fatal(err)
+			}
+			// wait for webhook config to take effect
+			time.Sleep(5 * time.Second)
```

Comment:
this asks for flakes. You have to poll you test below until the annotation is visible.